src/nbl/asset/interchange/CPLYMeshWriter.cpp

```diff
@@ -152,6 +152,8 @@ bool CPLYMeshWriter::writeAsset(io::IWriteFile* _file, const SAssetWriteParams&
 
     const void* ind = meshBuffer->getIndices();
     auto idxCnt = meshBuffer->getIndexCount();  // when you convert triangle Fan or triangle strip to triangle list, the index count changes, and thats what you should derive your face count from
+    // github comment
```

Comment:
todo

Reply:
as I can see in someone's code index count changes following formula `const size_t outputSize = _idxCount = (_idxCount - 2) * 3;` according to vulkan docs and it creates new buffer for Triangle Lists primitive topology. Either `idxBufferFromTrianglesFanToTriangles` and `idxBufferFromTrianglesFanToTriangles` use above formula to move vertices to a new buffer being created and `faceCount` variable is initialized as `faceCount = buf->getSize() / (idxtype == asset::EIT_16BIT ? 2u : 4u) / 3u;`, so I can see no reference to old `idxCnt`

Reply:
`faceCount = buf->getSize() / (idxtype == asset::EIT_16BIT ? 2u : 4u) / 3u;`
seems ok to me
https://github.com/Devsh-Graphics-Programming/Nabla/pull/54/files#diff-31af0c63cf74f1d0b3003ba1c3fa9f8b57801e5ab5dd68e0450cb342bfdf7676R174

Reply:
input mesh can be any topology

this code doesnt account for triangle strip, triangle fan, etc.

The plethora of static methods in `IMeshManipulator` for getting polycounts and getting triangles exists for a reason!

Reply:
> when you convert triangle Fan or triangle strip to triangle list, the index count changes, and thats what you should derive your face count from

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/54/files/e09503d06963541ab68393939d21d8c44cd750a2#r571410313 and that's why it is being converted I guess, but the line with `faceCount` uses buffer with Triangle Lists topoloy for sure to calculate itself. I will use the method anyways, I didn't know about it before

Reply:
> this code doesnt account for triangle strip, triangle fan, etc.

it does, first idx buffer is converted to triangle and from that face count is computed

Reply:
the "shitshow" actually starts here
https://github.com/Devsh-Graphics-Programming/Nabla/blob/e09503d06963541ab68393939d21d8c44cd750a2/src/nbl/asset/interchange/CPLYMeshWriter.cpp#L97

basically from this point onwards its wrong because assuming triangle list topology

Reply:
oh ok, didnt notice this, i thought we're talking about line 174

Reply:
yeah, it's a piece of shit, but it then gets updated and is fine

Reply:
anyway, I will use the method